examples/Examples/Common/Example.cs

```diff
@@ -0,0 +1,77 @@
+using CommandLine;
+
+namespace Concordium.Sdk.Examples.Common;
+
+/// <summary>
+/// Represents an example program which supports the command-line
+/// parameters defined by instances of <see cref="ExampleOptions"/>.
+/// </summary>
+public static class Example
+{
+    /// <summary>
+    /// Run an example program specified by a callback and the raw command-
+    /// line parameters. The raw command line parameters are parsed according
+    /// to the supplied <typeparam name="T"/> and the callback is invoked with
+    /// the resulting instance as its argument.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The <see cref="ExampleOptions"/> instance into which <paramref name="args"/>
+    /// will be parsed.
+    /// </typeparam>
+    /// <param name="args">
+    /// The raw command line arguments.
+    /// </param>
+    /// <param name="exampleCallback">
+    /// The callback corresponding to the example program which will be
+    /// invoked with the parsed <typeparam name="T"/> instance as its
+    /// argument.
+    /// </param>
+    public static void Run<T>(string[] args, Action<T> exampleCallback)
```

Comment:
I think this method is a bit over engineered.

IMHO I would make a code flow like this


```

public class GetBlockItemSummaryOptions
{
    [Option(HelpText = "Transaction hash to lookup", Required = true)]
    public string TransactionHash { get; set; }

    [Option(HelpText = "URL representing the endpoint where the gRPC V2 API is served.", Required = true)]
    public string Endpoint { get; set; }
}

public static class ExampleHelpers
{
    public static bool TryParse<T>(string[] args, out T? options)
    {
        var parserResult = Parser.Default
            .ParseArguments<T>(args);
        if (parserResult.Tag == ParserResultType.NotParsed)
        {
            foreach (var error in parserResult.Errors)
            {
                Console.WriteLine(error);
            }
            options = default;
            return false;
        }

        options = parserResult.Value;
        return true;
    }
}

public static class Program
{
    /// <summary>
    /// An example showing how one can query transaction status from a node.
    /// </summary>
    /// <param name="args">GetBlockItemSummaryOptions
    /// Example: --endpoint http://node.testnet.concordium.com/ --transactionhash 143ca4183d0bb204000ad08e0fd5792985c808861b97f3b81cb9016ad39d09d2</param>
    public static async Task Main(string[] args)
    {
        if (!ExampleHelpers.TryParse(args, out GetBlockItemSummaryOptions? options))
        {
            return;
        }

        var endpoint = new Uri(options!.Endpoint);
        var transactionHash = TransactionHash.From(options.TransactionHash);

        using var client = new ConcordiumClient(endpoint, 20_000);
…..
```

There would be a static helper class, but all runnable are defined within the main method. Callbacks are avoided and we stick to Task usages.

Reply:
I think there are drawbacks of adding boilerplate to the examples. IMO, we should just have a runnable example which details the API of the SDK.

Furthermore, we want the examples to behave similarly. Inheriting the options allow for this. Also, I believe it is wise to specify default values where sensisble since they are at least reflected in the help message, but in many cases are useful (e.g. localhost and port 20000). We should also be able to specify the port.